internal/declarative/v2/test/v1/declarative_test.go

```diff
@@ -127,12 +143,222 @@ var _ = Describe(
+	})
+
+	AfterAll(func() {
+		cancel()
+		Expect(env.Stop()).To(Succeed())
+	})
+})
+
+var _ = Describe("Test Manifest Reconciliation for module deletion", Ordered, func() {
+	var ctx context.Context
+	var cancel context.CancelFunc
+	var reconciler *Reconciler
+	var env *envtest.Environment
+	var cfg *rest.Config
+	var testClient client.Client
+
+	runID := fmt.Sprintf("run-%s", rand.String(4))
+	obj := &testv1.TestAPI{Spec: testv1.TestAPISpec{ManifestName: "deletion-manifest"}}
+	obj.SetLabels(labels.Set{testRunLabel: runID})
+	obj.SetNamespace(customResourceNamespace.Name)
+	obj.SetName(runID)
+
+	key := client.ObjectKeyFromObject(obj)
+
+	opts := []Option{WithRemoteTargetCluster(
+		func(context.Context, Object) (*ClusterInfo, error) {
+			return &ClusterInfo{
+				Config: cfg,
+			}, nil
+		},
+	)}
+	source := WithSpecResolver(DefaultSpec(filepath.Join(testSamplesDir, "raw-manifest.yaml"), RenderModeRaw))
+	oldDeployedResources, err := internal.ParseManifestToObjects(path.Join(testSamplesDir, "raw-manifest.yaml"))
+	Expect(err).NotTo(HaveOccurred())
+
+	BeforeAll(func() {
+		env, cfg = StartEnv()
+		testClient = GetTestClient(cfg)
+		ctx, cancel = context.WithCancel(context.TODO())
+		reconciler = StartDeclarativeReconcilerForRun(ctx, runID, cfg, append(opts, WithSpecResolver(source))...)
+	})
+
+	It("Should create manifest resources", func() {
+		Expect(testClient.Create(ctx, obj)).To(Succeed())
+
+		EventuallyDeclarativeStatusShould(
+			ctx, key, testClient,
+			BeInState(StateReady),
+			HaveConditionWithStatus(ConditionTypeResources, metav1.ConditionTrue),
+			HaveConditionWithStatus(ConditionTypeInstallation, metav1.ConditionTrue),
+		)
+
+		Expect(testClient.Get(ctx, key, obj)).To(Succeed())
+		Expect(obj.GetStatus()).To(HaveAllSyncedResourcesExistingInCluster(ctx, testClient))
+	})
+
+	It("Should remove deployed module resources after its deletion", func() {
+		source := WithSpecResolver(DefaultSpec(filepath.Join(testSamplesDir, "empty-file.yaml"), RenderModeRaw))
+		reconciler.SpecResolver = source
+		Eventually(func() error {
+			for _, res := range oldDeployedResources.Items {
+				currentRes := &unstructured.Unstructured{}
+				currentRes.SetGroupVersionKind(res.GroupVersionKind())
+				currentRes.SetName(res.GetName())
+				currentRes.SetNamespace(customResourceNamespace.Name)
+				err := testClient.Get(ctx, client.ObjectKeyFromObject(currentRes), currentRes)
+				if !k8serrors.IsNotFound(err) {
+					return ErrOldResourcesStillDeployed
+				}
+			}
+			return nil
+		}, testutils.Timeout, testutils.Interval).WithContext(ctx).Should(Succeed())
+
+	})
+
+	It("Should remove module resources from status.synced", func() {
+		EventuallyDeclarativeStatusShould(
+			ctx, key, testClient,
+			BeInState(StateReady),
+			HaveConditionWithStatus(ConditionTypeResources, metav1.ConditionTrue),
+			HaveConditionWithStatus(ConditionTypeInstallation, metav1.ConditionTrue),
+		)
+
+		Eventually(func() error {
```

Comment:
Same as above, I suggest defining a separate function and pass it as an argument when asserting with `Eventually` for better readability.